verticapy/toolbox.py

```diff
@@ -1610,3 +1610,300 @@ def __round__(self, x):
+    for p in prefix:
+        n = len(p)
+        if p in name_tmp and name_tmp[:n] == p:
+            name_tmp = name_tmp[n:]
+            break
+    return name_tmp
+
+# ---#
+def erase_suffix_in_name(name: str, suffix: list = []):
+    """
+-------------------------------------------------------------------------------
+Excludes the input lists of suffixes from the input name and returns it.
+When there is a match, the other elements of the list are ignored.
+
+Parameters
+---------- 
+name: str
+    Input name.
+suffix: list, optional
+    List of suffixes.
+
+Returns
+-------
+name
+    The name without the suffixes.
+    """
+    name_tmp = name
+    for s in suffix:
+        n = len(s)
+        if s in name_tmp and name_tmp[-n:] == s:
+            name_tmp = name_tmp[:-n]
+            break
+    return name_tmp
+
+# ---#
+def erase_word_in_name(name: str, word: list = []):
+    """
+-------------------------------------------------------------------------------
+Excludes the input lists of words from the input name and returns it.
+When there is a match, the other elements of the list are ignored.
+
+Parameters
+---------- 
+name: str
+    Input name.
+word: list, optional
+    List of words.
+
+Returns
+-------
+name
+    The name without the input words.
+    """
+    for w in word:
+        if w in name:
+            return name.replace(w, "")
+            break
+    return name
+
+# ---#
+def erase_in_name(
+    name: str, 
+    suffix: list = [], 
+    prefix: list = [],
+    word: list = [],
+    order: list = ["p", "s", "w"]
+):
+    """
+-------------------------------------------------------------------------------
+Excludes the input lists of suffixes and prefixes from the input name and 
+returns it. When there is a match, the other elements of the list are ignored.
+
+Parameters
+---------- 
+name: str
+    Input name.
+suffix: list, optional
+    List of suffixes.
+prefix: list, optional
+    List of prefixes.
+word: list, optional
+    List of words.
+order: list, optional
+    The order of the process.
+        s: suffix
+        p: prefix
+        w: word
+    For example the list ["p", "s", "w"] will start by excluding the 
+    prefixes, then suffixes and finally the input words.
+
+Returns
+-------
+name
+    The name without the prefixes, suffixes and input words.
+    """
+    name_tmp = name
+    f = {"p": (erase_prefix_in_name, prefix), "s": (erase_suffix_in_name, suffix), "w": (erase_word_in_name, word)}
+    for x in order:
+        name_tmp = f[x][0](name_tmp, f[x][1])
+    return name_tmp
+
+# ---#
+def is_similar_name(
+    name1: str, 
+    name2: str, 
+    skip_suffix: list = [], 
+    skip_prefix: list = [],
+    skip_word: list = [],
+    order: list = ["p", "s", "w"]
+):
+    """
+-------------------------------------------------------------------------------
+Excludes the input lists of suffixes, prefixes and words from the input name 
+and returns it.
+
+Parameters
+---------- 
+name1: str, optional
```

Comment:
i dont think mae1 name2 are optional as the function below does erase does not have the name as optional,